index.js

```diff
@@ -225,6 +228,17 @@ style.registerProtocols = function(tilelive) {
   tilelive.protocols["tmstyle:"] = this;
 };
 
+function resolveTileliveUri(fromUri, toUri) {
+  // Normalize uris
+  fromUri = url.parse(fromUri);
+  toUri = url.parse(toUri);
+
+  // Url.parse reads the first ".." as the hostname
+  const toUriPathName = path.join(toUri.hostname, toUri.path);
+
+  return toUri.protocol + '//' + path.resolve(fromUri.pathname, toUriPathName);
```

Comment:
This is only appropriate for sources that read from the filesystem. In many cases, the protocol may be `http:`, `mapbox:`, or `tilejson+http(s):`, in which case relative resolution doesn't make sense.

Off the top of my head, appropriate protocols are `tmsource:`, `mbtiles:`, `tilejson:`, `file:`, and `<whatever>+file:` (possibly `bridge:`), so these should probably be whitelisted.

https://github.com/mojodna/tilelive-modules/blob/master/index.json contains a mostly complete list of modules to refer to.